poly/src/polynomial/univariate/sparse.rs

```diff
@@ -44,6 +47,12 @@ impl<F: Field> core::ops::Deref for SparsePolynomial<F> {
     }
 }
 
+impl<F: Field> core::ops::DerefMut for SparsePolynomial<F> {
+    fn deref_mut(&mut self) -> &mut [(usize, F)] {
+        &mut self.coeffs
+    }
+}
```

Comment:
Why do we need this? Isn't this an implementation detail?

Reply:
I looked at the code. I think this is to be consistent with DensePolynomial, which already has the two handy implementations like this one:

```
impl<F: Field> Deref for DensePolynomial<F> {
    type Target = [F];

    fn deref(&self) -> &[F] {
        &self.coeffs
    }
}

impl<F: Field> DerefMut for DensePolynomial<F> {
    fn deref_mut(&mut self) -> &mut [F] {
        &mut self.coeffs
    }
}
```

It does simplify some coding.

Reply:
This might be a useful thing to keep.

Reply:
The impl for `DensePolynomial` makes sense, as the natural implementation of a polynomial would be as a list of coefficients. IMO for sparse polynomials it's less clear, because you can have different "natural" implementations: a list of (index, coeff) pairs like right now, or a Hashmap<usize, F>, or maybe another implementation.

Reply:
Got it. Then I think the best way is to get rid of `DerefMut` for `SparsePolynomial` for now.

Reply:
Actually, this is a problem---we have to expose an interface for the data inside `SparsePolynomial`, otherwise `DensePolynomial` would be unable to add a sparse polynomial. 

A possibility is `Iterator` instead of `Deref` or `DeferMut`, where  `SparsePolynomial` should be able to allow one to iterate over all its key-value pairs.

Reply:
Maybe we can just add an `Index` impl which returns the `i`-th coefficient? And a corresponding `fn get(&self, index: usize) -> Option<&F>` method which returns the coefficient? This is analogous to the methods on slices.

Reply:
Would that be slightly slow? Since one now has to iterate all the possible indices (up to the degree).

Reply:
I see what you mean. Hm let's just keep this as is.